src/udev/cdrom_id/cdrom_id.c

```diff
@@ -1047,6 +727,95 @@ static int open_drive(Context *c) {
         return 0;
 }
 
+static const char * const feature_table[_FEATURE_MAX] = {
+        [FEATURE_RW_NONREMOVABLE] = "RW_NONREMOVABLE",
+        [FEATURE_RW_REMOVABLE]    = "RW_REMOVABLE",
+
+        [FEATURE_MO_SE]           = "MO_SE",
+        [FEATURE_MO_WO]           = "MO_WO",
+        [FEATURE_MO_AS]           = "MO_AS",
+
+        [FEATURE_CD_ROM]          = "CD",
+        [FEATURE_CD_R]            = "CD_R",
+        [FEATURE_CD_RW]           = "CD_RW",
+
+        [FEATURE_DVD_ROM]         = "DVD",
+        [FEATURE_DVD_R]           = "DVD_R",
+        [FEATURE_DVD_RAM]         = "DVD_RAM",
+        [FEATURE_DVD_RW_RO]       = "DVD_RW_RO",
+        [FEATURE_DVD_RW_SEQ]      = "DVD_RW_SEQ",
+        [FEATURE_DVD_R_DL_SEQ]    = "DVD_R_DL_SEQ",
+        [FEATURE_DVD_R_DL_JR]     = "DVD_R_DL_JR",
+        [FEATURE_DVD_RW_DL]       = "DVD_RW_DL",
+        [FEATURE_DVD_R_DDR]       = "DVD_R_DDR",
+        [FEATURE_DVD_PLUS_RW]     = "DVD_PLUS_RW",
+        [FEATURE_DVD_PLUS_R]      = "DVD_PLUS_R",
+
+        [FEATURE_DDCD_ROM]        = "DDCD",
+        [FEATURE_DDCD_R]          = "DDCD_R",
+        [FEATURE_DDCD_RW]         = "DDCD_RW",
+
+        [FEATURE_DVD_PLUS_RW_DL]  = "DVD_PLUS_RW_DL",
+        [FEATURE_DVD_PLUS_R_DL]   = "DVD_PLUS_R_DL",
+
+        [FEATURE_BD]              = "BD",
+        [FEATURE_BD_R_SRM]        = "BD_R_SRM",
+        [FEATURE_BD_R_RRM]        = "BD_R_RRM",
+        [FEATURE_BD_RE]           = "BD_RE",
+
+        [FEATURE_HDDVD]           = "HDDVD",
+        [FEATURE_HDDVD_R]         = "HDDVD_R",
+        [FEATURE_HDDVD_RAM]       = "HDDVD_RAM",
+        [FEATURE_HDDVD_RW]        = "HDDVD_RW",
+        [FEATURE_HDDVD_R_DL]      = "HDDVD_R_DL",
+        [FEATURE_HDDVD_RW_DL]     = "HDDVD_RW_DL",
+
+        [FEATURE_MRW]             = "MRW",
+        [FEATURE_MRW_W]           = "MRW_W",
+};
```

Comment:
given that the feature enum is not precisely packed but has quite large values, it might make sense to not use the feature as index here, but instead have an array of `struct { enum … feature; const char *name }` in which the feature id is searched? could even use `bsearch()` then to translate the feature id to a string, which would mean we save a lot of static memory and runtime just deteriorates from O(1) to O(log(n)) which would be more than good enough